src/core/instance/state.js

```diff
@@ -141,7 +141,14 @@ function initData (vm: Component) {
         `Use prop default value instead.`,
         vm
       )
-    } else if (!isReserved(key)) {
+    } else if (isReserved(key)) {
+      warn(
```

Comment:
warn must be shown only in development mode (process.env.NODE_ENV !== production)

Reply:
fixed